File: MatrixMultiply.java


public class MatrixMultiply {
    public static void main(String[] args) {
        int n = Integer.parseInt(args[0]);
        int[][] m1 = makeMatrix(n);
        int[][] m2 = makeMatrix(n);
//        printMatrix(m1);
//        System.out.println();
//        printMatrix(m2);
//        System.out.println();
        
        int[][] m3 = multiply(m1, m2, n);
//        System.out.println();
//        System.out.println();
        printMatrix(m3);
    }
    
    private static void printMatrix(int[][] m) {
    	for(int[] x :  m) {
        	System.out.print("[");
        	for(int y : x) {
        		System.out.print(y + " ");
        	}
        	System.out.println("]");
        }
    }

    private static int[][] makeMatrix(int n) {
        int[][] m3 = new int[n][n];
        for (int i = 0; i < n; i++) {
            for(int j = 0; j < n; j++) {
                m3[i][j] = i + j;
            }
        }

        return m3;
    }
    
    private static int[][] multiply(int[][] m1, int[][]m2, int n) {
    	int[][] m3 = new int[n][n];
    	
    	for(int i = 0; i < n; i++) {
    		for(int j = 0; j < n; j++) {
    			int[] m1_array = m1[i];
    			int[] m2_array = new int[n];
    			
    			int count = 0;
    			for(int[] k : m2) {
    				m2_array[count++] = k[j];
    			}
    			
    			
//    			System.out.print("\n\nm1:\n[");
//    			for(int x: m1_array) {
//    				System.out.print(x + " ");
//    			}
//    			System.out.println("]");
//    			System.out.print("m2:\n[");
//    			for(int x: m2_array) {
//    				System.out.print(x + " ");
//    			}
//    			System.out.println("]\n");
    			
    			int matrix_result = dot_product(m1_array, m2_array, n);
    			
    			m3[i][j] = matrix_result;
    		}
    	}
    	return m3;
    }
    
    private static int dot_product(int[] arr1, int[] arr2, int n) {
    	int result = 0;
    	for(int i = 0; i < n; i++) {
    		result += arr1[i] * arr2[i];
    	}
    	return result;
    	
    }

}

File: matrix-multiply.py
import sys

#Jancy Capellan
#Zachary Katz


def main():
    n = int(sys.argv[1])
    # m1 = [[1,2,3], [4,5,6], [7,8,9]]
    # m2 = [[1,2,1], [3,1,1], [5,1,4]]
    m1 = makeMatrix(n)
    m2 = makeMatrix(n)

    m3 = multiply(m1, m2, n)
    for i in m3:
        print("[", end="")
        for j in i:
            print(str(j) + " ", end="")
        print("]")
    # print(multiply(m1, m2, n))


def makeMatrix(n):
    m3 = []
    for i in range(n):
        m3.append([])
        for j in range(n):
            m3[i].append(i+j)
    
    return m3

def multiply(m1, m2, n):
    m3 = []
    for i in range(n):
        m3.append([])


    #Loop through m1 rows (i: 0 to n exlusive)
    for i in range(n):
        #Loop through m2 columns (j: 0 to n exlusive)
        for j in range(n):
            #Get two arrays, call dot product
            m1_array = m1[i]
            m2_array = []
            for k in m2:
                m2_array.append(k[j])

            matrix_result = dot_product(m1_array, m2_array, n)
            #store result in (i, j) of m3
            m3[i].append(matrix_result)
            
    return m3

def dot_product(arr1, arr2, n):
    result = 0
    for i in range(n):
        result += arr1[i]*arr2[i]
    return result





if __name__ == "__main__":
    main()